subsys/usb/usb_transfer.c

```diff
@@ -313,8 +313,22 @@ int usb_transfer_sync(uint8_t ep, uint8_t *data, size_t dlen, unsigned int flags
 		return ret;
 	}
 
-	/* Semaphore will be released by the transfer completion callback */
-	k_sem_take(&pdata.sem, K_FOREVER);
+	/* Semaphore will be released by the transfer completion callback
+	 * which might not be called when transfer was cancelled
+	 */
+	while (1) {
+		struct usb_transfer_data *trans;
+
+		ret = k_sem_take(&pdata.sem, K_MSEC(100));
```

Comment:
Making this configurable is probably too much, but could you please add a define for this, `#define USB_TRANSFER_SYNC_TIMEOUT    100`?

Reply:
done, actually I did not think about the timeout value too much, 100ms is just first try